teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -14,113 +14,138 @@ authors:
+- [Question: without using persistent volume can i share workspace among tasks?](https://github.com/tektoncd/pipeline/issues/3704#issuecomment-980748302):
+This user uses an NFS for their workspace to avoid provisioning a PVC on every PipelineRun.
+- [FR: Allow volume from volumeClaimTemplate to be used in more than one workspace](https://github.com/tektoncd/pipeline/issues/3440):
+This issue highlights usability concerns with using the same PVC in multiple workspaces (done using sub-paths).
+
+## Existing Workarounds and Mitigations
+
+There's currently no workaround that addresses the overhead of extra pods or storage without harming reusability.
+
+### Combine multiple pieces of functionality in one Task
+Instead of combining functionality provided by Tasks into Pipelines, a Task or Pipeline author could use Steps or a multifunctional script to combine
+all necessary functionality into a single Task. This allows multiple "actions" to be run in one pod, but hurts reusability and makes parallel execution more difficult.
+
+### Use PipelineResources (deprecated) to express a workflow in one Task
+PipelineResources allowed multiple pieces of functionality to run in a single pod by building some of these functions into the TaskRun controller.
+This allowed some workflows to be written as single Tasks.
+For example, the "git" and "image" PipelineResources made it possible to create a workflow that cloned and built a repo, and pushed the resulting image to
+an image repository, all in one Task. 
+However, PipelineResources hurt reusability because they required Task authors to anticipate what other functionality would be needed before and after the Task.
+For this reason, among others, they were deprecated; please see [TEP-0074](./0074-deprecate-pipelineresources.md) for more information.
+
+### Rely on the Affinity Assistant for improved TaskRun scheduling
+The Affinity Assistant schedules TaskRuns that share a PVC on the same node.
+This feature allows TaskRuns that share PVCs to run in parallel in a system that supports only `ReadWriteOnce` Persistent Volumes.
+However, this does not address the underlying issues of pod overhead and the need to shuttle data between TaskRuns in different pods.
+It also comes with its own set of drawbacks, which are described in more detail in
+[TEP-0046: Colocation of Tasks and Workspaces](https://github.com/tektoncd/community/pull/318/files).
+
+### Use Task results to share data without using PVCs
+Tasks may emit string results that can be used as [parameters of subsequent Tasks](https://tekton.dev/docs/pipelines/pipelines/#passing-one-task-s-results-into-the-parameters-or-when-expressions-of-another).
+There is an existing [TEP](https://github.com/tektoncd/community/pull/477/files) for supporting dictionary and array results as well.
+However, results are not designed to handle large, arbitrary forms of data like source repositories or images.
```

Comment:
in the context of https://github.com/tektoncd/community/pull/521 (and https://hackmd.io/a6Kl4oS0SaOyBqBPTirzaQ) there have been some suggestions to make it possible to produce arbitrarily large results (and potentially merge workspaces and results into one concept) - however even that would still not address the underlying problem of needing to move the data around between pods (even arbitrarily large results would still need to backed by _something_ outside of the pod)